tests/src/test/scala/com/nvidia/spark/rapids/StringFallbackSuite.scala

```diff
@@ -29,7 +29,17 @@ class StringFallbackSuite extends SparkQueryCompareTestSuite {
     "RegExpReplace",
     nullableStringsFromCsv, execsAllowedNonGpu = Seq("ProjectExec", "Alias",
       "RegExpReplace", "AttributeReference", "Literal")) {
-    frame => frame.selectExpr("regexp_replace(strings,null,'D')")
+    frame => {
+      ShimLoader.getSparkShims.getSparkShimVersion match {
+        case SparkShimVersion(major, minor, _) =>
+          // this test is not valid in Spark 3.1 and later because the expression is
+          // NullIntolerant and gets replaced with a null literal instead
+          val isValidTestForSparkVersion = major <= 3 && minor == 0
```

Comment:
Although it seems unnecessary to have this separate variable, it makes the output message more obvious:

```
Test Canceled: isValidTestForSparkVersion was false
```